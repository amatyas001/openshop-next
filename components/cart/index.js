export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
