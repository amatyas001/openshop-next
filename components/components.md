Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
