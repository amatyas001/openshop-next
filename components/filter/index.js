export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
