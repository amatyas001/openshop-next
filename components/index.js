export * from './cart';
export * from './filter';
export * from './product';
export * from './layout';
export * from './payment';
export * from './shared';
