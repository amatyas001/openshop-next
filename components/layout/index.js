export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
