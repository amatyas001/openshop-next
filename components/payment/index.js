export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
