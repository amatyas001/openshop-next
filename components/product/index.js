export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
