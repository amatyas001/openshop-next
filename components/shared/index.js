export * from './Button/Button';
export * from './ButtonIcon/ButtonIcon';
export * from './Field/Field';
export * from './Image/Image';
export * from './Panel/Panel';
export * from './Spinner/Spinner';
