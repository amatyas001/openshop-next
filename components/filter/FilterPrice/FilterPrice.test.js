import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import { FilterPrice } from '@app/components';
import { INITIAL_STATE } from '@app/config';
import { filterPrice } from '@app/lib/redux/actions';
import { mockProduct } from '@app/__mocks__/@app/mocks';

const mockStore = createStore([]);

const products = mockProduct(5).map((c, i) => {
  return {
    ...c,
    price: i * 15,
  };
});

describe('<FilterPrice />', () => {
  let tree, store, slider, current;

  beforeAll(() => {
    store = mockStore(INITIAL_STATE);

    act(() => {
      tree = create(
        <Provider store={store}>
          <FilterPrice products={products} />
        </Provider>
      );
    });

    slider = tree.root.findByProps({
      'data-testid': 'filter-price-slider',
    });

    current = tree.root.findByProps({
      'data-testid': 'filter-price-current',
    });
  });

  it('should render with required props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should set maximum value', () => {
    expect(slider.props.max).toEqual((products.length - 1) * 15);
  });

  it('should handle price filter', async () => {
    const mock_value = 10;

    expect(current.props.children).toContain(slider.props.max);

    await act(async () => {
      await slider.props.onChange(mock_value);
      slider.props.onMouseUp();
    });

    expect(store.getActions()).toEqual([filterPrice(mock_value)]);
    expect(slider.props.value).toEqual(mock_value);
    expect(current.props.children).toContain(mock_value);
  });
});
