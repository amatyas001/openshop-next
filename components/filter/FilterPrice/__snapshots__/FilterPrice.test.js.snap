// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPrice /> should render with required props 1`] = `
<div
  className="css-gx0lhm"
>
  <label
    className="css-1m5v70x"
    htmlFor="price"
  >
    Price
  </label>
  <span
    className="css-1pmifls"
    data-testid="filter-price-current"
  >
    60
     $
  </span>
  <div
    className="css-ftmy5t"
    data-testid="filter-price-slider"
    onBlur={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role="presentation"
    tabIndex="-1"
  >
    <div
      aria-label="price-track"
      className="css-jscp3i"
      data-slider-track=""
    />
    <div
      aria-label="price-filled"
      className="css-g3jgex"
      data-slider-filled-track=""
    />
    <div
      aria-label="price-thumb"
      aria-orientation="horizontal"
      aria-valuemax={60}
      aria-valuemin={1}
      aria-valuenow={60}
      className="css-1974d3q"
      data-slider-thumb=""
      onKeyDown={[Function]}
      role="slider"
      tabIndex={0}
    />
    <input
      name="price"
      type="hidden"
      value={60}
    />
  </div>
</div>
`;
